src/endpoint/sts/ops/sts_post_assume_role.js

```diff
@@ -21,10 +22,28 @@ async function assume_role(req) {
         }
         throw new StsError(StsError.InternalFailure);
     }
+
+    let seconds_until_expiration;
+    // Check if duration_seconds was provided via the CLI, otherwise resort to the default
+    if (req.body.duration_seconds) {
+        // duration_seconds is a string, convert it to a number
+        const duration_seconds = Number(req.body.duration_seconds);
+        // If it was provided, check if it's within the allowed range dictated by Amazon
+        if (duration_seconds < 900 || duration_seconds > 43200) {
+            dbg.error('Value', duration_seconds, 'for duration_seconds is not within the allowed range of 900-43200 seconds');
+            throw new StsError(StsError.ValidationError);
+        }
+        seconds_until_expiration = duration_seconds;
+    } else {
+        seconds_until_expiration = Math.ceil(config.STS_DEFAULT_SESSION_TOKEN_EXPIRY_MS / 1000);
+    }
```

Comment:
Please take it to another function (or functions) that would parse and validate the input and return what you need (in this case `expiration`).